### PR TITLE
[9.0] fix: useLegacyAdapter should check the VO

### PR DESCRIPTION
**Pull request by aldbr**
Aims at solving https://github.com/DIRACGrid/DIRAC/issues/7346

BEGINRELEASENOTES
*Configuration
FIX: useLegacyAdapter now checks the VO
ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/ConfigurationSystem/Client/PathFinder.py [hunk 1]
```diff
@@ -256,9 +256,20 @@ def useLegacyAdapter(system, service=None) -> bool:
 
     :return: bool -- True if DiracX should be used
     """
+    from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+
+    # Check if DiracX is enabled for this service
     system, service = divideFullName(system, service)
     value = gConfigurationData.extractOptionFromCFG(f"/DiracX/LegacyClientEnabled/{system}/{service}")
-    return (value or "no").lower() in ("y", "yes", "true", "1")
+    isServiceEnabled = (value or "no").lower() in ("y", "yes", "true", "1")
```
Comment (chrisburr) [inline src/DIRAC/ConfigurationSystem/Client/PathFinder.py]: How about returning quickly and not calling `getProxyInfo` if it's pointless:

```suggestion
    if (value or "no").lower() not in ("y", "yes", "true", "1"):
        return False
```

File: src/DIRAC/ConfigurationSystem/Client/PathFinder.py [hunk 2]
```diff
@@ -256,9 +256,20 @@ def useLegacyAdapter(system, service=None) -> bool:
...
+    # Check if DiracX is enabled for this VO
+    result = getProxyInfo()
```
Comment (chrisburr) [inline src/DIRAC/ConfigurationSystem/Client/PathFinder.py:274]: What happens if we're using a host certificate rather than a proxy?

Comment (aldbr) [reply src/DIRAC/ConfigurationSystem/Client/PathFinder.py:274]: How does `diracx` handle requests coming from a host?

**Comment by chrisburr** [comment]
See https://github.com/DIRACGrid/DIRAC/issues/7346

**Pull request closed without merging**